sdk/dotnet/Makefile

```diff
@@ -1,19 +1,44 @@
-PROJECT_NAME       := Pulumi .NET Core SDK
-LANGHOST_PKG       := github.com/pulumi/pulumi/sdk/dotnet/cmd/pulumi-language-dotnet
-VERSION            := $(shell ../../scripts/get-version)
-VERSION_DOTNET     := ${VERSION:v%=%}                             # strip v from the beginning
-VERSION_PREFIX     := $(firstword $(subst -, ,${VERSION_DOTNET})) # e.g. 1.5.0
-VERSION_SUFFIX     := $(word 2,$(subst -, ,${VERSION_DOTNET}))    # e.g. alpha.1
-PROJECT_PKGS       := $(shell go list ./cmd...)
-PULUMI_LOCAL_NUGET ?= ~/.nuget/local
+PROJECT_NAME         := Pulumi .NET Core SDK
+LANGHOST_PKG         := github.com/pulumi/pulumi/sdk/dotnet/cmd/pulumi-language-dotnet
+
+PROJECT_PKGS         := $(shell go list ./cmd...)
+PULUMI_LOCAL_NUGET   ?= ~/.nuget/local
+
+VERSION              := $(shell ../../scripts/get-version --embed-feature-branch)
+VERSION_DOTNET       := ${VERSION:v%=%}                                   # strip v from the beginning
+VERSION_FIRST_WORD   := $(strip $(word 1,$(subst -, ,${VERSION_DOTNET}))) # e.g. 1.5.0
+VERSION_SECOND_WORD  := $(strip $(word 2,$(subst -, ,${VERSION_DOTNET}))) # e.g. alpha or alpha.1
+VERSION_THIRD_WORD   := $(strip $(word 3,$(subst -, ,${VERSION_DOTNET}))) # e.g. featbranch or featbranch.1
+
+VERSION_PREFIX       := ${VERSION_FIRST_WORD}
+
+ifeq ($(VERSION_THIRD_WORD),)
+	VERSION_SUFFIX   := ${VERSION_SECOND_WORD}
+else
+	# have to use addsuffix or make adds a whitespace between the two values.
+	VERSION_SUFFIX   := $(addsuffix -$(VERSION_THIRD_WORD),${VERSION_SECOND_WORD})
+endif
 
 TESTPARALLELISM := 10
 
 include ../../build/common.mk
 
 build::
-	dotnet build dotnet.sln /p:VersionPrefix=${VERSION_PREFIX} /p:VersionSuffix=${VERSION_SUFFIX}
+	# .NET support is only in Preview currently.  So unilaterally add `-preview` to the version suffix.
+	# From the nuget docs:
+	#
+	# Pre-release versions are then denoted by appending a hyphen and a string after the patch number.
+	# Technically speaking, you can use any string after the hyphen and NuGet will treat the package as
+	# pre-release. NuGet then displays the full version number in the applicable UI, leaving consumers
+	# to interpret the meaning for themselves.
+	#
+	# With this in mind, it's generally good to follow recognized naming conventions such as the
+	# following:
+	#
+	#     -alpha: Alpha release, typically used for work-in-progress and experimentation
+	dotnet build dotnet.sln /p:VersionPrefix=${VERSION_PREFIX} /p:VersionSuffix=preview-${VERSION_SUFFIX}
 	echo "Copying NuGet packages to ${PULUMI_LOCAL_NUGET}"
+	rm -rf ${PULUMI_LOCAL_NUGET}
```

Comment:
this is scary.  but in line with what we do in other SDKs.  effectively, we blow away the 'local installation' location, recreate it, and then copy the fresh built values into it.

Reply:
We should probably not default to ` ~/.nuget/local/`, then. Why not put these in `/opt/pulumi` like everything else?

Reply:
> We should probably not default to ~/.nuget/local/, then. Why not put these in /opt/pulumi like everything else?

@mikhailshilkov ?  Would that work?  Can nuget pick up from `/opt/pulumi` instead of `.nuget/local`?

Reply:
Ah, it looks like it would.  looks like we manually do this in tests with `localNuget = filepath.Join(usr.HomeDir, ".nuget", "local")`.  I'll see about making the change here.

Reply:
Except there's no `/opt/pulumi` on Windows...